File: sdks/registry-go/examples/minimal_register.go
package main

import (
	"context"
	"log"
	"time"

	registrysdk "github.com/rinfra/rinfra/sdks/registry-go/registrysdk"
)

func main() {
	client, err := registrysdk.NewClient(
		registrysdk.Config{
			MainAddress:  "127.0.0.1:7946",
			ClusterToken: "change-me-in-production",
			PingInterval: 2 * time.Second,
		},
		registrysdk.Registration{
			NodeID: "go-order-api-1",
			Endpoints: []registrysdk.Endpoint{
				{Protocol: "http", Address: "10.0.1.27:8080"},
			},
			Metadata: map[string]string{
				"service.name":        "order-api",
				"service.instance_id": "go-order-api-1",
				"service.version":     "0.1.0",
				"service.env":         "dev",
			},
		},
	)
	if err != nil {
		log.Fatalf("new client: %v", err)
	}

	if err := client.Start(context.Background()); err != nil {
		log.Fatalf("start client: %v", err)
	}

	time.Sleep(20 * time.Second)

	nodes, err := client.ListNodes(context.Background())
	if err == nil {
		log.Printf("node count: %d", len(nodes))
	}

	if err := client.Stop(context.Background()); err != nil {
		log.Fatalf("stop client: %v", err)
	}
}

File: sdks/registry-java/examples/MinimalRegister.java
package io.rinfra.registry.sdk.examples;

import io.rinfra.registry.sdk.Endpoint;
import io.rinfra.registry.sdk.Registration;
import io.rinfra.registry.sdk.RegistryClient;
import io.rinfra.registry.sdk.RegistryClientConfig;

import java.util.List;
import java.util.Map;

public final class MinimalRegister {
    public static void main(String[] args) throws Exception {
        RegistryClientConfig config = new RegistryClientConfig("127.0.0.1:7946", "change-me-in-production");
        Registration registration = new Registration(
                "java-order-api-1",
                List.of(new Endpoint("http", "10.0.1.23:8080")),
                Map.of(
                        "meta.schema", "rinfra.meta.v1",
                        "service.name", "order-api",
                        "service.instance_id", "java-order-api-1",
                        "service.version", "0.1.0",
                        "service.env", "dev"
                )
        );

        RegistryClient client = new RegistryClient(config, registration);
        client.start();

        Runtime.getRuntime().addShutdownHook(new Thread(client::stop));
        Thread.sleep(60_000);
        client.stop();
    }
}
